app/src/main/java/io/inai/android_sample_integration/drop_in/PresentCheckoutFragment.kt

```diff
@@ -0,0 +1,144 @@
+            NetworkRequestHandler.KEY_AUTH_STRING to authenticationString,
+            NetworkRequestHandler.KEY_POST_DATA_JSON to Json.encodeToString(orderPostData)
+        )
+        makeNetworkRequest(networkConfig, ::onOrderPrepared)
+    }
+
+    private fun onOrderPrepared(orderResponse: String) {
+        hideProgress()
+        val orderResult = json.decodeFromString<OrderResult>(orderResponse)
+        Config.customerId = orderResult.customer_id
+        orderId = orderResult.id
+        Log.d("ORDER_ID", orderId)
+        initializeCheckout()
+    }
+
+    private fun initializeCheckout() {
+        if (Config.inaiToken.isNotEmpty() && orderId.isNotEmpty()) {
+            val config = InaiConfig(
+                token = Config.inaiToken,
+                orderId = orderId,
+                countryCode = Config.countryCode,
+                redirectUrl = ""
+            )
+            try {
+                val inaiCheckout = InaiCheckout(config)
+                inaiCheckout.presentCheckout(
+                    requireContext(),
+                    this
+                )
+            } catch (ex: Exception) {
+                //  Handle initialisation error
+                requireActivity().showAlert("Error while initialising sdk : $ex.message")
+            }
+        }
+    }
+
+    override fun paymentFinished(result: InaiPaymentResult) {
+        when (result.status) {
+            InaiPaymentStatus.Success -> {
+                requireActivity().showAlert("Payment Success! ${result.data}")
+            }
+            InaiPaymentStatus.Failed -> {
+                requireActivity().showAlert("Payment Failed! ${result.data}")
+            }
+            InaiPaymentStatus.Canceled -> {
+                var message = "Payment Canceled!"
+                if (result.data.has("message")) {
+                    message = result.data.getString("message")
+                }
+                requireActivity().showAlert(message)
+            }
+        }
+    }
+
+    /**
+     *  Helper Functions
+     */
+    private fun makeNetworkRequest(
+        networkConfig: Map<String, String>,
+        callback: (String) -> Unit
+    ) {
+        NetworkRequestHandler.makeNetworkRequest(networkConfig) { result: NetworkRequestHandler.Result ->
+            when (result) {
+                is NetworkRequestHandler.Result.Success -> {
+                    callback(result.message)
+                }
+                is NetworkRequestHandler.Result.Failure -> {
+                    onError(result.message)
+                }
+            }
+        }
+    }
+
+    private fun getDataForOrders(): OrderPostData {
+        return OrderPostData(
+            amount = Config.amount,
+            currency = Config.currency,
+            customer = OrderCustomer(
+                email = "testdev@inai.io",
+                first_name = "Dev",
+                last_name = "Smith",
+                contact_number = "01010101010",
```

Comment:
Can we make this consistent across too? Just for uniformity sake,
```suggestion
                email = "customer@example.com",
                first_name = "John",
                last_name = "Doe",
                contact_number = "01010101010",
```

Reply:
Done